src/libraries/System.Data.Common/src/System/Data/Common/DbDataReaderExtensions.cs

```diff
@@ -43,12 +46,12 @@ private void PopulateFields()
                 IsUnique = GetDbColumnValue<bool?>(SchemaTableColumn.IsUnique);
                 NumericPrecision = GetDbColumnValue<int?>(SchemaTableColumn.NumericPrecision);
                 NumericScale = GetDbColumnValue<int?>(SchemaTableColumn.NumericScale);
-                UdtAssemblyQualifiedName = GetDbColumnValue<string>("UdtAssemblyQualifiedName");
-                DataType = GetDbColumnValue<Type>(SchemaTableColumn.DataType);
-                DataTypeName = GetDbColumnValue<string>("DataTypeName");
+                UdtAssemblyQualifiedName = GetDbColumnValue<string?>("UdtAssemblyQualifiedName");
+                DataType = GetDbColumnValue<Type?>(SchemaTableColumn.DataType);
+                DataTypeName = GetDbColumnValue<string?>("DataTypeName");
             }
 
-            private T GetDbColumnValue<T>(string columnName) => _schemaColumns.Contains(columnName) && _schemaRow[columnName] is T value ? value : (default);
+            private T GetDbColumnValue<T>(string columnName) => _schemaColumns.Contains(columnName) && _schemaRow[columnName] is T value ? value : default!;
```

Comment:
I assume when this is used by `GetDbColumnValue<string>(SchemaTableColumn.ColumnName)` that we'll never hit the `default!` case?

Reply:
It's actually possible (and valid) for this to return null, but I can't annotate it as such with `[MaybeNull]` because it's a local method. I don't think there are any negative consequences as this is private anyway and the end result is the same...